src/lib/network/clients/network_proposal.rb

```diff
@@ -24,24 +29,79 @@ def description
 
     def make_proposal(_)
       {
-        "preformatted_proposal" => Yast::Lan.Summary("proposal"),
-        "label_proposal"        => [Yast::LanItems.summary("one_line")]
+        "preformatted_proposal" => preformatted_proposal,
+        "label_proposal"        => [Yast::LanItems.summary("one_line")],
+        "links"                 => BACKEND_LINKS
       }
     end
 
     def ask_user(args)
-      log.info "Launching network configuration"
-      begin
-        Yast::Wizard.OpenAcceptDialog
+      result =
+        case args["chosen_id"]
+        when "network--switch-to-wicked"
+          switch_to_wicked
+        when "network--switch-to-nm"
+          switch_to_network_manager
+        else
+          launch_network_configuration(args)
+        end
 
-        result = Yast::WFM.CallFunction("inst_lan", [args.merge("skip_detection" => true)])
+      { "workflow_sequence" => result }
+    end
+
+  private
+
+    def preformatted_proposal
+      proposal_text = switch_backend_link
+      proposal_text.prepend(Yast::Lan.Summary("proposal")) if wicked_backend?
+      proposal_text
+    end
 
-        log.info "Returning from the network configuration with: #{result}"
-      ensure
-        Yast::Wizard.CloseDialog
+    def switch_backend_link
+      if wicked_backend?
+        info     = _("Using Wicked")
+        switcher = Hyperlink(SWITCH_TO_NETWORK_MANAGER, _("switch to Network Manager"))
+      else
+        info     = _("Using Network Manager")
+        switcher = Hyperlink(SWITCH_TO_WICKED, _("switch to Wicked"))
       end
 
-      { "workflow_sequence" => result }
+      "<ul><li>#{info} (#{switcher})</li></ul>"
+    end
+
+    def launch_network_configuration(args)
```

Comment:
I usually use the extra spaces (empty lines) to ease the reading of method when there is related/unrelated "group of lines". Do you know what I mean? I also like to left an empty line before the return value.

That said, I'll to remove them if you think that it is a better approach, but it would be nice to achieve an agreement about that :)

Reply:
BTW the latest Rubocop has some new checks related to the spacing so maybe it will force a decision here. :wink: